tools/sdk-versioner/src/main.rs

```diff
@@ -0,0 +1,288 @@
+                crate_path
+                    .as_os_str()
+                    .to_str()
+                    .expect("valid utf-8 path")
+                    .to_string(),
+            ),
+        );
+    }
+
+    // Set the `version` if one was given
+    if opt.sdk_version.is_some() {
+        value.insert(
+            "version".to_string(),
+            Value::String(
+                if is_sdk_crate {
+                    &opt.sdk_version
+                } else {
+                    &opt.smithy_version
+                }
+                .clone()
+                .unwrap(),
+            ),
+        );
+    }
+}
+
+/// Recursively discovers Cargo.toml files in the given `path` and adds them to `manifests`.
+fn discover_manifests(manifests: &mut Vec<PathBuf>, path: impl AsRef<Path>) -> anyhow::Result<()> {
+    let path = path.as_ref();
+
+    for entry in fs::read_dir(path)? {
+        let entry = entry?;
+        if entry.path().is_dir() {
+            discover_manifests(manifests, entry.path())?;
+        } else if entry.path().is_file()
+            && entry.path().file_name() == Some(OsStr::new("Cargo.toml"))
+        {
+            manifests.push(entry.path().into());
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{update_dependencies, Opt};
+    use pretty_assertions::assert_eq;
+    use toml::Value;
+
+    const TEST_MANIFEST: &[u8] = br#"
+        [package]
+        name = "test"
+        version = "0.1.0"
+
+        [dependencies]
+        aws-config = "0.4.1"
+        aws-sdk-s3 = "0.4.1"
+        aws-smithy-types = "0.34.1"
+        aws-smithy-http = { version = "0.34.1", features = ["test-util"] }
+        something-else = "0.1"
+    "#;
+
+    #[track_caller]
+    fn test_with_opt(opt: Opt, expected: &[u8]) {
+        let mut metadata: Value = toml::from_slice(TEST_MANIFEST).unwrap();
+        assert!(update_dependencies(
+            metadata
+                .get_mut("dependencies")
+                .unwrap()
+                .as_table_mut()
+                .unwrap(),
+            &opt,
+        )
+        .unwrap());
+
+        let expected: Value = toml::from_slice(expected).unwrap();
+        assert_eq!(expected, metadata);
+    }
+
+    #[test]
+    fn update_dependencies_with_versions() {
+        test_with_opt(
+            Opt {
+                crate_paths: Vec::new(),
+                sdk_path: None,
+                sdk_version: Some("0.5.0".to_string()),
+                smithy_version: Some("0.35.0".to_string()),
+            },
+            br#"
+            [package]
+            name = "test"
+            version = "0.1.0"
+
+            [dependencies]
+            aws-config = { version = "0.5.0" }
+            aws-sdk-s3 = { version = "0.5.0" }
+            aws-smithy-types = { version = "0.35.0" }
+            aws-smithy-http = { version = "0.35.0", features = ["test-util"] }
+            something-else = "0.1"
+            "#,
+        );
+    }
```

Comment:
testing with inline `Cargo.toml`s is good but it's probably worth adding a filesystem-based test with [tempfile](https://docs.rs/tempfile/latest/tempfile/) like I did for the [sync tool fs operations](https://github.com/awslabs/smithy-rs/blob/21ffa905dbc048dde6dff654bf953ee01df478b5/tools/smithy-rs-sync/src/fs.rs#L217)